### DISCUSSION TITLE
What happens if all redis nodes aren't available

### DISCUSSION THREAD

**Discussion by DeniVuMedi** [brainix/pottery#452] [Q&A] [answered]
Hi,

I'm currently using pottery for redlock purposes with 1 redis master.
I'm aware 5 are needed to eliminate a single point of failure but as it happens I'm bound to 1 node.

My question is, what's the expected behaviour should the node fail or be unavailable?
The following will cause my code to hang forever on trying to acquire the lock:
```python
redis = Redis.from_url("redis://wrong_url")
lock = Redlock(key="name", masters={redis})
with lock:
    print("Never prints")
print("Never prints")
```

I think the appropriate behaviour is to either raise an exception or skip everything covered by the lock?
Just blocking the process isn't good?

**Comment by DeniVuMedi** [comment, marked as answer]
Didn't mean to answer my own question but I just noticed the optional parameter `raise_on_redis_errors`.
Btw I couldn't figure out what `context_manager_timeout` does.

**Comment by raulFuzita** [comment]
if you're using `Redis` or `StrictRedis`, you should set the timeout in their respective configurations. If you're using `ConnectionPool`, ensure the timeout is set there instead.
You set `socket_connect_timeout` for initial connection and `socket_timeout` for commands Read/Write. It'll throw an exception when Redlock tries to stablish a connection.

Example:
```python
import redis

redis.ConnectionPool.from_url('redis://localhost:6379/0', socket_connect_timeout=2, socket_timeout=3)
redis.StrictRedis(connection_pool=self._pool)

Redlock(key="name", masters={redis}, raise_on_redis_errors=True)
```
